pyscf/adc/radc.py

```diff
@@ -1697,6 +1701,35 @@ def get_trans_moments(adc):
     T = np.array(T)
     return T
 
+def density_matrix(adc, T):
+
+    nocc = adc._nocc
+    nvir = adc._nvir
+
+    n_singles = nocc
+    n_doubles = nvir * nocc * nocc
+    ij_ind = np.tril_indices(nocc, k=-1)
+
+    s1 = 0
+    f1 = n_singles
+    s2 = f1
+    f2 = s2 + n_doubles
+
+    T_doubles = T[:,n_singles:]
+    T_doubles = T_doubles.reshape(-1,nvir,nocc,nocc)
+    T_doubles_transpose = T_doubles.transpose(0,1,3,2).copy()
+    T_bab = (2/3)*T_doubles + (1/3)*T_doubles_transpose
+
+    T_aaa = T_bab - T_bab.transpose(0,1,3,2)
+
+    T_a = T[:,s1:f1]
+    T_bab = T_bab.reshape(-1,n_doubles)
+    T_aaa = T_aaa.reshape(-1,n_doubles)
+
+    dm = 2 * np.dot(T_a,T_a.T) + np.dot(T_aaa, T_aaa.T) + 2 * np.dot(T_bab, T_bab.T)
+
+    return dm
```

Comment:
Here it's OK to return a total 1-RDM, because it's spin-adapted ADC, but not in the UADC case.